files/bin/setup-local-disks

```diff
@@ -0,0 +1,206 @@
+
+print_help() {
+  echo "usage: $0 <raid0 | mount>"
+  echo "Sets up Amazon EC2 Instance Store NVMe disks"
+  echo ""
+  echo "-d, --dir directory to mount the filesystem(s) (default: /mnt/k8s-disks/)"
+  echo "-h, --help print this help"
+}
+
+# Sets up a RAID-0 of NVMe instance storage disks, moves
+# the contents of /var/lib/kubelet and /var/lib/containerd
+# to the new mounted RAID, and bind mounts the kubelet and
+# containerd state directories.
+maybe_raid0() {
+  local md_name="kubernetes"
+  local md_device="/dev/md/${md_name}"
+  local md_config="/.aws/mdadm.conf"
+  local array_mount_point="${MNT_DIR}/0"
+  mkdir -p "$(dirname "${md_config}")"
+
+  if [[ ! -s "${md_config}" ]]; then
+    mdadm --create --force --verbose \
+      "${md_device}" \
+      --level=0 \
+      --name="${md_name}" \
+      --raid-devices="${#EPHEMERAL_DISKS[@]}" \
+      "${EPHEMERAL_DISKS[@]}"
+    while [ -n "$(mdadm --detail "${md_device}" | grep -ioE 'State :.*resyncing')" ]; do
+      echo "Raid is resyncing..."
+      sleep 1
+    done
+    mdadm --detail --scan > "${md_config}"
+  fi
+
+  ## Check if the device symlink has changed on reboot to include a homehost identifier
+  local current_md_device=$(find /dev/md/ -type l -regex '.*kubernetes_?[0-9a-z]*$' | tail -n1)
+  if [[ ! -z ${current_md_device} ]]; then
+    md_device="${current_md_device}"
+  fi
+
+  # Format the array if not already formatted.
+  if [[ -z "$(lsblk "${md_device}" -o fstype --noheadings)" ]]; then
+    ## By default, mkfs tries to use the stripe unit of the array (512k),
+    ## for the log stripe unit, but the max log stripe unit is 256k.
+    ## So instead, we use 32k (8 blocks) to avoid a warning of breaching the max.
+    ## mkfs.xfs defaults to 32k after logging the warning since the default log buffer size is 32k.
+    mkfs.xfs -l su=8b "${md_device}"
+  fi
+
+  ## Create the mount directory
+  mkdir -p "${array_mount_point}"
+
+  local dev_uuid=$(blkid -s UUID -o value "${md_device}")
+  local mount_unit_name="$(systemd-escape --path --suffix=mount "${array_mount_point}")"
+  cat > "/etc/systemd/system/${mount_unit_name}" << EOF
+  [Unit]
+  Description=Mount EC2 Instance Store NVMe disk RAID0
+  [Mount]
+  What=UUID=${dev_uuid}
+  Where=${array_mount_point}
+  Type=xfs
+  Options=defaults,noatime
+  [Install]
+  WantedBy=multi-user.target
+EOF
+  systemd-analyze verify "${mount_unit_name}"
+  systemctl enable "${mount_unit_name}" --now
+
+  prev_running=""
+  needs_linked=""
+  for unit in "containerd" "kubelet"; do
+    ## Check if the bind mount from the RAID already exists
+    if [[ "$(systemctl is-active var-lib-${unit}.mount)" != "active" ]]; then
+      # Check if components that depend on the RAID are running and, if so, stop them
+      if systemctl is-active "${unit}" > /dev/null 2>&1; then
+        prev_running+=" ${unit}"
+      fi
+      needs_linked+=" ${unit}"
+    fi
+  done
+
+  if [[ ! -z "${prev_running}" ]]; then
+    systemctl stop ${prev_running}
```

Comment:
This will stop `containerd` before `kubelet` if they're both running, because of the order of the values on line 81?